content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/vault-queries.md

```diff
@@ -0,0 +1,561 @@
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+After the output state has been finalized it will be represented as the following in the database (`custom_representation column`):
+
+```json
+{
+  "net.corda.v5.ledger.utxo.ContractState" : {
+    "stateRef": "<TransactionID>:<StateIndex>"
+  },
+  "com.r3.corda.demo.contract.TestState" : {
+    "testField": ""
+  } 
+}
+```
+
+{{< note >}}
+The `net.corda.v5.ledger.utxo.ContractState` field is a part of the JSON representation for all state types.
+{{< /note >}}
+{{< note >}}
+The implementation of the JSON factory needs to be defined in the same CPK as the state
+it is working on so the platform can get hold of it when persisting a state to the database.
+{{< /note >}}
+
+Use this representation to create the vault-named queries in the next section.
+
+## Creating and Registering a Vault-Named Query
+
+Registration means the query is stored on sandbox creation time and can be executed later on.
+Vault-named queries need to be part of a contract CPK. Corda installs the vault named query when the contract CPK is uploaded.
+To create and register a query the `net.corda.v5.ledger.utxo.query.VaultNamedQueryFactory` interface must be implemented.
+
+### Basic Vault-Named Query Registration Example
+
+A simple vault-named query implementation for this `TestState` would look like this:
+
+{{< note >}}
+This example has no filtering, mapping, or collecting logic.
+{{< /note >}}
+
+{{< tabs name="tabs-3" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+class DummyCustomQueryFactory : VaultNamedQueryFactory {
+    override fun create(vaultNamedQueryBuilderFactory: VaultNamedQueryBuilderFactory) {
+        vaultNamedQueryBuilderFactory.create("DUMMY_CUSTOM_QUERY")
+            .whereJson(
+                "WHERE visible_states.custom_representation -> 'com.r3.corda.demo.contract.TestState' " +
+                        "->> 'testField' = :testField"
+            )
+            .register()
+    }
+}
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+
+```java
+public class DummyCustomQueryFactory implements VaultNamedQueryFactory {
+    @Override
+    public void create(@NotNull VaultNamedQueryBuilderFactory vaultNamedQueryBuilderFactory) {
+        vaultNamedQueryBuilderFactory.create("DUMMY_CUSTOM_QUERY")
+                .whereJson(
+                        "WHERE visible_states.custom_representation -> 'com.r3.corda.demo.contract.TestState' " +
+                                "->> 'testField' = :testField"
+                )
+                .register();
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+The interface called `VaultNamesQueryFactory` has one method:
+
+`void create(@NotNull VaultNamedQueryBuilderFactory vaultNamedQueryBuilderFactory);`
+
+This function is called upon start-up and should define how a query will operate in it with the following steps:
+
+* To create a vault-named query with a given name, in this case it is `DUMMY_CUSTOM_QUERY`, call `vaultNamedQueryBuilderFactory.create()`.
```

Comment:
Just check that this is correct:

in this case it is `DUMMY_CUSTOM_QUERY`, call `vaultNamedQueryBuilderFactory.create()`.